collector/lib/EnvVar.h

```diff
@@ -25,7 +26,7 @@ class EnvVar {
   const T& value() const {
     std::call_once(init_once_, [this]() {
       const char* env_val = std::getenv(env_var_name_);
-      if (!env_val || !*env_val) {
+      if (!env_val) {
```

Comment:
This clause makes an empty env var (defined but length==0) result in the default value being used.

With the addition of the string-list parsing, I would like to parse an empty value as an empty list instead.

Since the Boolean env var is the only pre-existing type, I implemented the empty->default fallback as a special case. See https://github.com/stackrox/collector/blob/463b71885a9170a48c5a18a1e64aace908fcf8de/collector/lib/EnvVar.h#L53-L55